clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -202,7 +203,7 @@ public boolean matches(AbstractRequest body) {
 
         sender.run(time.milliseconds());  // Send AddOffsetsRequest
         assertTrue(transactionManager.hasPendingOffsetCommits());  // We should now have created and queued the offset commit request.
-        assertFalse(addOffsetsResult.isDone());
+        assertTrue(addOffsetsResult.isCompleted());
```

Comment:
Actually I may be wrong here. It seems the intent was to chain the TxnOffsetCommit onto the AddOffsets so that the the latter only completed when the former also finished. @apurvam can you clarify?